frontend/model/contracts/identity.js

```diff
@@ -1,10 +1,19 @@
 'use strict'
 
 import sbp from '@sbp/sbp'
+
 import { Vue, L } from '@common/common.js'
 import { merge } from './shared/giLodash.js'
 import { objectOf, objectMaybeOf, arrayOf, string, object } from '~/frontend/model/contracts/misc/flowTyper.js'
-import { noUppercase } from '~/frontend/views/utils/validators.js'
+import {
+  allowedUsernameCharacters,
+  noConsecutiveHyphensOrUnderscores,
+  noLeadingOrTrailingHyphen,
+  noLeadingOrTrailingUnderscore,
+  noUppercase
+} from '~/frontend/views/utils/validators.js'
```

Comment:
If code is (1) included in the slim version of contracts (and this code should be), and (2) is shared with the frontend UI code, then it belongs somewhere inside the `frontend/model/contracts/shared/` folder.

This helps us remember to think carefully when making modifications to the behavior of such code, so that if the code changes, we can make sure that the usage in the UI (usually but not always the latest version of such code) doesn't conflict with whatever version of the contract that's being used.

See `frontend/model/contracts/shared/README.md` for details.

Reply:
@snowteamer This was marked resolved but doesn't seem to be?

Reply:
@taoeffect Not sure what needed to be done here. Did you mean the `validarors.js` file should be moved to `contracts/shared/`?

Reply:
@snowteamer yep!